src/transactions/TransactionFrame.cpp

```diff
@@ -1629,10 +1650,13 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
             if (protocolVersionStartsFrom(ledgerVersion,
                                           SOROBAN_PROTOCOL_VERSION))
             {
-                // If transaction fails, we don't charge for any
-                // refundable resources.
-                mFeeRefund = declaredSorobanResourceFee() -
-                             mSorobanResourceFee->non_refundable_fee;
+                if (isSoroban())
```

Comment:
`isSoroban` should be moved to the check above as we don't want to add diagnostic events to the classic txs. It's ok though to postpone this to my change since I'm regenerating the tx meta anyway, though there is still this weird breakage in my PR, so maybe it's better to regenerate it for your (smaller) PR.